react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+
+
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {displayIndex: 0, displayId: this.props.array[0].id};
```

Comment:
You really only need one thing in state, either the `id` or the `index` of the active image